CHANGELOG.md

```diff
@@ -2,6 +2,23 @@
 
 ## Unreleased
 
+  - Encourage passing the `props` object as the first argument to components.
+
+    Components are just functions so this isn't mandatory and you can still
+    define arguments as you see fit.  The pattern of passing `props` makes
+    composition easier, encourages code which is more readable and decouples
+    the implementation of the component from the action of connecting it to the
+    store.  From now on, connecting will only work for components with zero
+    arity or which take an object as the first argument.
+
+  - `connect` now takes a selector as an optional second argument.
+
+    You can now pass an optional selector function to `connect`.  It will be
+    passed the `state` and should return an object which will be merged with
+    the component's `props` using `Object.assign({}, props, substate)`.
```

Comment:
The order of first `props` then `substate` mimics what Redux does. I've wondered why they implement it this way. `substate, props` would allow overriding the props from state which sounds like it could be useful sometimes.

Reply:
I definitely think this makes more sense but I am curious as to what trade off the redux team made when they designed it the other way. There's definitely a reason behind it.